libs/domain/checkout/src/mixins/checkout.mixin.ts

```diff
@@ -1,14 +1,44 @@
-import { Type } from '@spryker-oryx/di';
+import { resolve, Type } from '@spryker-oryx/di';
 import {
-  ComponentMixin,
-  ContentComponentProperties,
-} from '@spryker-oryx/experience';
+  ConnectableSignal,
+  signal,
+  signalAware,
+} from '@spryker-oryx/utilities';
 import { LitElement } from 'lit';
+import { map } from 'rxjs';
+import { CheckoutProcessState } from '../models';
+import { CheckoutService } from '../services';
 
-export const CheckoutComponentMixin = <T>(): Type<
-  LitElement & ContentComponentProperties<T>
-> => {
-  class CheckoutComponent extends ComponentMixin<T>() {}
+export declare class CheckoutMixinInterface {
+  protected checkoutService: CheckoutService;
 
-  return CheckoutComponent as Type<LitElement & ContentComponentProperties<T>>;
+  /**
+   * Indicates that the checkout is ready for collecting checkout data.
+   */
+  protected isAvailable: ConnectableSignal<boolean>;
+  protected isBusy: ConnectableSignal<boolean>;
+}
+
+export const CheckoutMixin = <T extends Type<LitElement>>(
+  superClass: T
+): Type<CheckoutMixinInterface> & T => {
+  @signalAware()
+  class CheckoutMixinClass extends superClass {
+    protected checkoutService = resolve(CheckoutService);
+
+    protected isAvailable = signal(
+      this.checkoutService
+        .getProcessState()
+        .pipe(map((state) => state !== CheckoutProcessState.NotAvailable)),
```

Comment:
`.pipe(map((state) => state !== CheckoutProcessState.Ready)),` ?

Reply:
A busy checkout is still available.